src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/ViewContext.cs

```diff
@@ -116,7 +117,8 @@ public ViewContext()
 
             if (viewData == null)
             {
-                throw new ArgumentNullException(nameof(viewData));
+                var metadata = viewContext.HttpContext.RequestServices.GetRequiredService<IModelMetadataProvider>();
+                viewData = new ViewDataDictionary(metadata);
             }
```

Comment:
Actually, this fallback is either unnecessary or incorrect. Since this copy constructor is only used internally for `HtmlHelper` and `ViewViewComponentResult` scenarios where the passed `ViewDataDictionary` should never be `null`, I lean toward the first. (Am I missing a scenario?) Otherwise the VDD should be created from the one in the passed `ViewContext`.